src/main/java/io/vertx/mqtt/impl/MqttServerConnection.java

```diff
@@ -87,8 +90,13 @@ void handleMessage(Object msg) {
 
       DecoderResult result = mqttMessage.decoderResult();
       if (result.isFailure()) {
-        chctx.pipeline().fireExceptionCaught(result.cause());
-        return;
+        Throwable cause = result.cause();
+        if (cause instanceof MqttUnacceptableProtocolVersionException) {
+          //
```

Comment:
my idea was about creating the endpoint here and sending the message not change the overall logic of handleConnect. Doesn't it work easily?